piq/vsi.py

```diff
@@ -0,0 +1,285 @@
+    x_lab = rgb2lab(x, data_range=data_range)
+
+    x_fft = torch.fft(torch.stack([x_lab, torch.zeros_like(x_lab)], dim=-1), 2)
+    lg = log_gabor(size_to_use, omega_0, sigma_f).to(x_fft).view(1, 1, *size_to_use, 1)
+    x_ifft_real = torch.ifft(x_fft * lg, 2)[..., 0]
+    s_f = x_ifft_real.pow(2).sum(dim=1).sqrt()
+
+    coordinates = torch.stack((torch.meshgrid([torch.arange(0, size_to_use[0]) - (size[0] - 1) / 2,
+                                               torch.arange(0, size_to_use[1]) - (size[1] - 1) / 2])), dim=0).to(x)
+    s_d = torch.exp(-torch.sum(coordinates, dim=0) / sigma_d ** 2)
+
+    eps = torch.finfo(x_lab.dtype).eps
+    min_x = x_lab.min(dim=-1, keepdim=True).values.min(dim=-2, keepdim=True).values
+    max_x = x_lab.max(dim=-1, keepdim=True).values.max(dim=-2, keepdim=True).values
+    normilized = (x_lab - min_x) / (max_x - min_x + eps)
+
+    norm = normilized[:, 1:].pow(2).sum(dim=1)
+    s_c = 1 - torch.exp(-norm / sigma_c ** 2)
+
+    vs_m = interpolate((s_f * s_d * s_c).unsqueeze(1), size[-2:], mode='bilinear', align_corners=True).squeeze(1)
+
+    min_vs_m = vs_m.min(dim=-1, keepdim=True).values.min(dim=-2, keepdim=True).values
+    max_vs_m = vs_m.max(dim=-1, keepdim=True).values.max(dim=-2, keepdim=True).values
+    return (vs_m - min_vs_m) / (max_vs_m - min_vs_m + eps)
+
+
+def rgb2xyz(x: torch.Tensor) -> torch.Tensor:
+    """
+    Translates image from RGB color space to XYZ
+    Input image in [0, 1] range.
+    """
+
+    mask_below = x <= 0.04045
+    mask_above = x > 0.04045
+
+    tmp = x / 12.92 * mask_below + torch.pow((x + 0.055) / 1.055, 2.4) * mask_above
+
+    weights_RGB_to_XYZ = torch.tensor([[0.4124564, 0.3575761, 0.1804375],
+                                       [0.2126729, 0.7151522, 0.0721750],
+                                       [0.0193339, 0.1191920, 0.9503041]]).to(x)
+
+    x_xyz = torch.matmul(tmp.permute(0, 2, 3, 1), weights_RGB_to_XYZ.t()).permute(0, 3, 1, 2)
+
+    return x_xyz
+
+
+def xyz2lab(x: torch.Tensor) -> torch.Tensor:
+    epsilon = 0.008856
+    kappa = 903.3
+    illuminants_d50 = torch.tensor([0.9642119944211994, 1., 0.8251882845188288]).to(x).view(1, 3, 1, 1)
+
+    tmp = x / illuminants_d50
+
+    mask_below = tmp <= epsilon
+    mask_above = tmp > epsilon
+    tmp = torch.pow(tmp, 1. / 3.) * mask_above + (kappa * tmp + 16.) / 116. * mask_below
+
+    weights_XYZ_to_LAB = torch.tensor([[0, 116., 0],
+                                       [500., -500., 0],
+                                       [0, 200., -200.]]).to(x)
+    bias_XYZ_to_LAB = torch.tensor([-16., 0., 0.]).to(x).view(1, 3, 1, 1)
+
+    x_lab = torch.matmul(tmp.permute(0, 2, 3, 1), weights_XYZ_to_LAB.t()).permute(0, 3, 1, 2) + bias_XYZ_to_LAB
+    return x_lab
+
+
+def rgb2lab(x: torch.Tensor, data_range: float = 255.) -> torch.Tensor:
+    return xyz2lab(rgb2xyz(x / data_range))
+
+
+def log_gabor(size: tuple, omega_0: float = 0.021, sigma_f: float = 1.34) -> torch.Tensor:
```

Comment:
Add `from typing import Tuple` on top

```suggestion
def log_gabor(size: Tuple[int, int], omega_0: float = 0.021, sigma_f: float = 1.34) -> torch.Tensor:
```